qutebrowser/browser/webengine/darkmode.py

```diff
@@ -236,6 +237,19 @@ def copy_add_setting(self, setting: _Setting) -> '_Definition':
         new._settings = self._settings + (setting,)  # pylint: disable=protected-access
         return new
 
+    def copy_replace_setting(self, old: _Setting, new: _Setting) -> '_Definition':
+        """Get a new _Definition object with `old` replaced by `new`.
+
+        If `old` is not in the settings list, return the old _Definition object.
+        """
+        if old not in self._settings:
+            return self
```

Comment:
Should be an error IMHO, if we don't want any chance, we wouldn't call this in the first place.

Reply:
I think it'd make sense, to turn a silent failure into a loud one. If you rewrite your `for` slightly to `break` when a match was found, this seems like a great use-case for Python's (somewhat exotic) [for/else](https://book.pythontips.com/en/latest/for_-_else.html).

Reply:
Funny, I was also thinking if for/else is applicable here.